Dependencies/xr/Source/Shared/XRHelpers.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <XR.h>
+
+namespace xr
+{
+    /**
+    * Compare two poses and determine if there is a meaningful difference between them.
+    **/
+    constexpr float FLOAT_COMPARISON_THRESHOLD = 0.02f;
+    bool PoseWasMeaningfullyUpdated(const Pose& lhs, const Pose& rhs)
+    {
+        return abs(lhs.Position.X - rhs.Position.X) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Position.Y - rhs.Position.Y) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Position.Z - rhs.Position.Z) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Orientation.X - rhs.Orientation.X) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Orientation.Y - rhs.Orientation.Y) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Orientation.Z - rhs.Orientation.Z) > FLOAT_COMPARISON_THRESHOLD
+            || abs(lhs.Orientation.W - rhs.Orientation.W) > FLOAT_COMPARISON_THRESHOLD;
+    }
```

Comment:
I'm not sure we want to be in the business of making decisions about what's a meaningful update or not; that seems like it should be either the platform's or the app's decision, not XR's.

Reply:
I want to revisit this later.  For now this is sane for both of the supported platforms and stops us from spamming with re-draws.  I agree that this makes sense to be pushed to the consuming app for more accurate devices.